src/cpp/src/decoder.cpp

```diff
@@ -224,30 +206,54 @@ void LinkPredictionDecoder::forward(Batch *batch, bool train) {
 }
 
 DistMult::DistMult() {
-    if (marius_options.training.loss_function_type == LossFunctionType::SoftMax) {
-        loss_function_ = new SoftMax();
-    } else if (marius_options.training.loss_function_type == LossFunctionType::RankingLoss) {
-        loss_function_ = new RankingLoss(marius_options.training.margin);
+    if (marius_options.loss.loss_function_type == LossFunctionType::SoftMax) {
```

Comment:
Can we move this into some function: `LossFunction *getLossFunction(LossFunctionType loss_function_type)`

That way we can reuse that function for all the Decoders